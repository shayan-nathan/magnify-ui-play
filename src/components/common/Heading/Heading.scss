// TODO: Sort them based on font-size

.c-heading {
  &--1 {
    color: $color-primary;
    font-size: 28px;
    line-height: 40px;
    font-weight: bold;
  }

  &--2 {
    color: $color-primary;
    font-size: 20px;
    line-height: 40px;
  }

  // login headings
  &--3 {
    color: $color-font-main;
    font-size: 28px;
    line-height: 40px;
    font-weight: normal;
  }

  &--4 {
    color: $color-font-main;
    font-size: 20px;
    font-weight: bold;
    line-height: 40px;
  }

  &--5 {
    color: $color-primary;
    font-weight: 600;
    font-size: 12px;
    line-height: 24px;
    margin-bottom: 8px;
  }

  // used in segment builder
  &--6 {
    color: rgba(0, 0, 0, 0.8);
    font-weight: 600;
    font-size: 18px;
    line-height: 24px;
    margin-bottom: 12px;
  }
}
