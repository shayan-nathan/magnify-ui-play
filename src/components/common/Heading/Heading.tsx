import classNames from 'classnames'
import { variant } from 'models'

export type headingLvl = '1' | '2' | '3' | '4' | '5' | '6'

interface Props {
  level: headingLvl
  variant: variant
  className?: string
  children: JSX.Element | string
}

export function Heading({ level, variant, className, children }: Props): JSX.Element {
  const Element = level !== '6' ? (`h${level}` as keyof JSX.IntrinsicElements) : `p`

  return (
    <Element className={classNames({ [`c-heading c-heading--${variant} ${className}`]: true })} data-testid='heading'>
      {children}
    </Element>
  )
}
