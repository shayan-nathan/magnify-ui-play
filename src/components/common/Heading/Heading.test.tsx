import { render, screen } from '@testing-library/react'

import { Heading } from 'components/common/Heading/Heading'

const component = (
  <Heading level='1' variant='1'>
    My Heading
  </Heading>
)
describe('<Heading />', () => {
  it('renderes with the correct content', () => {
    render(component)
    const element = screen.getByTestId('heading')

    expect(element).toMatchSnapshot()
    expect(element).toContainHTML('My Heading')
  })

  it('renders the right lvl of element', () => {
    render(component)
    const element = screen.getByTestId('heading')
    expect(element.tagName).toBe('H1')
  })

  it('renders the right lvl and variant', () => {
    render(
      <Heading level='2' variant='3'>
        My Heading
      </Heading>,
    )
    const h2Element = screen.getByTestId('heading')
    expect(h2Element.tagName).toBe('H2')
    expect(h2Element).toHaveClass('c-heading--3')
  })

  it('display a pharagraph if the lvl is too high', () => {
    render(
      <Heading level='6' variant='3'>
        My Heading
      </Heading>,
    )
    const h2Element = screen.getByTestId('heading')
    expect(h2Element.tagName).toBe('P')
    expect(h2Element).toHaveClass('c-heading--3')
  })
})
