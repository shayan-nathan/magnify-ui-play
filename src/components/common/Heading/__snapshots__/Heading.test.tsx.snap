// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renderes with the correct content 1`] = `
<h1
  class="c-heading c-heading--1 undefined"
  data-testid="heading"
>
  My Heading
</h1>
`;
